lib/metadataTypes/Asset.js

```diff
@@ -458,23 +490,88 @@ class Asset extends MetadataType {
         metadata.assetType.id = this.definition.typeMapping[metadata.assetType.name];
 
         // define asset's subtype
-        const subType = this.getSubtype(metadata);
+        const subType = this._getSubtype(metadata);
 
         // #1 get text extracts back into the JSON
         await this._mergeCode(metadata, deployDir, subType);
 
         // #2 get file from local disk and insert as base64
         await this._readExtendedFileFromFS(metadata, subType, deployDir);
 
+        // only execute #3 if we are deploying / copying from one BU to another, not while using mcdev as a developer tool
+        if (buObject.mid && metadata.memberId !== buObject.mid) {
+            // #3 make sure customer key is unique by prefixing it with target MID (unless we are deploying to the same MID)
+            if (!metadata[this.definition.keyField].startsWith(buObject.mid)) {
+                // check if this prefixed with the source MID
+                const suffix = '-' + buObject.mid;
```

Comment:
not sure if you shouldnt prefix instead of suffix. If the external key is too long this may be truncated and then you may get conflicts again

Reply:
my bad, missed your overview :)